src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -499,8 +499,34 @@ export class TerminalService implements ITerminalService {
 		this.onDidChangeInstances(() => updateTerminalContextKeys());
 	}
 
-	async getActiveOrCreateInstance(): Promise<ITerminalInstance> {
-		return this.activeInstance || this.createTerminal();
+	async getActiveOrCreateInstance(options?: { acceptsInput?: boolean }): Promise<ITerminalInstance> {
+		const activeInstance = this.activeInstance;
+		// No instance, create
+		if (!activeInstance) {
+			return this.createTerminal();
```

Comment:
what if someone's default profile does not accept input and `acceptsInput:true`?

Reply:
I don't think that's possible, a profile should always accept input?